repos/fdbt-site/src/pages/api/manageTimeRestriction.ts

```diff
@@ -0,0 +1,205 @@
+import isArray from 'lodash/isArray';
+import { NextApiResponse } from 'next';
+import { getTimeRestrictionByNameAndNoc, insertTimeRestriction } from '../../data/auroradb';
+import { removeAllWhiteSpace, removeExcessWhiteSpace, isValid24hrTimeFormat } from './apiUtils/validator';
+import { NextApiRequestWithSession, ErrorInfo, FullTimeRestriction, TimeBand } from '../../interfaces';
+import { redirectToError, redirectTo, getAndValidateNoc } from './apiUtils';
+import { updateSessionAttribute } from '../../utils/sessions';
+import { GS_TIME_RESTRICTION_ATTRIBUTE } from '../../constants/attributes';
+import { TimeRestrictionDay } from 'shared/matchingJsonTypes';
+
+export const collectInputsFromRequest = (
+    req: NextApiRequestWithSession,
+    timeRestrictionDays: TimeRestrictionDay[],
+): FullTimeRestriction[] => {
+    const fullTimeRestrictions: FullTimeRestriction[] = [];
+    timeRestrictionDays.forEach((day) => {
+        const startTimeInputs = req.body[`startTime${day}`];
+        const endTimeInputs = req.body[`endTime${day}`];
+        if (isArray(startTimeInputs) && isArray(endTimeInputs)) {
+            const timeBands = startTimeInputs.map((startTime, index) => {
+                return {
+                    startTime: removeAllWhiteSpace(startTime),
+                    endTime: removeAllWhiteSpace(endTimeInputs[index]),
+                };
+            });
+            fullTimeRestrictions.push({
```

Comment:
Rather than a forEach with a push to an array, could we change this to a map? I think it would be cleaner and easier to follow?